### PR TITLE
Kaichao implement clear command, clean code, fix integer out of bounds bug and tests

### PR DESCRIPTION
**Pull request by kaichaoang**

### CODE REVIEW

File: src/main/java/seedu/stock/logic/parser/AddCommandParser.java [hunk 1]
```diff
@@ -36,6 +36,7 @@ public AddCommand parse(String args) throws ParseException {
 
         if (!arePrefixesPresent(argMultimap, PREFIX_NAME, PREFIX_LOCATION, PREFIX_SOURCE, PREFIX_QUANTITY)
                 || !argMultimap.getPreamble().isEmpty()) {
+            System.out.println("thrown 1");
```
Comment (Ashley-Lau) [inline src/main/java/seedu/stock/logic/parser/AddCommandParser.java]: i think this is for you to debug and you forgot to remove?

File: src/test/java/seedu/stock/model/stock/AccumulatedQuantityTest.java [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+        AccumulatedQuantity incrementedAccumulatedQuantity = accumulatedQuantity
+                                                                    .getIncrementedAccumulatedQuantity();
+        AccumulatedQuantity expectedIncrementedAccumulatedQuantity = new AccumulatedQuantity("100");
+        assertNotEquals(accumulatedQuantity, incrementedAccumulatedQuantity);
+        assertEquals(incrementedAccumulatedQuantity, expectedIncrementedAccumulatedQuantity);
+    }
+
+    @Test
+    public void isValidAccumulatedQuantity() {
+        // null accumulated quantity
```
Comment (Ashley-Lau) [inline src/test/java/seedu/stock/model/stock/AccumulatedQuantityTest.java:38]: minor inconsistency in the naming

File: src/test/java/seedu/stock/model/stock/AccumulatedQuantityTest.java [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+        AccumulatedQuantity accumulatedQuantity = new AccumulatedQuantity("100");
+        AccumulatedQuantity accumulatedQuantityCopy = new AccumulatedQuantity("100");
+        AccumulatedQuantity differentAccumulatedQuantity = new AccumulatedQuantity("99");
+
+
+        // returns true
+        assertTrue(accumulatedQuantity.equals(accumulatedQuantity)); // compare to self
+        assertTrue(accumulatedQuantity.equals(accumulatedQuantityCopy)); // compare to a copy of itself
+
+        // returns false, different accumulated quantity
```
Comment (Ashley-Lau) [inline src/test/java/seedu/stock/model/stock/AccumulatedQuantityTest.java:72]: here too

**Review by Ashley-Lau — APPROVED** [review]
Minor issues to be solved. LGTM after resolving these issues

File: src/test/java/seedu/stock/testutil/UpdateStockDescriptorBuilder.java [hunk 4]
```diff
@@ -13,7 +13,7 @@
 import seedu.stock.model.stock.Stock;
 
 /**
- * A utility class to help with building EditPersonDescriptor objects.
+ * A utility class to help with building EditStockDescriptor objects.
```
Comment (AudreyFelicio) [inline src/test/java/seedu/stock/testutil/UpdateStockDescriptorBuilder.java]: Should be `UpdateStockDescriptor`

**Review by AudreyFelicio — APPROVED** [review]
LGTM overall besides some minor naming issues.

**Pull request merged**